### ISSUE TITLE
timeouts for downloads

### ISSUE THREAD

**Issue by chreman** [ContentMine/getpapers#58]
I regularly get timeouts and retries when using e.g. `-x`
that does not happen when batching requests e.g. through the query (into months). is there a possibility to include automatic batching for larger query results?

**Comment by Blahah** [comment]
Yes, that seems sensible. In general the handling of timeouts etc. needs a bit of work.

**Comment by chreman** [comment]
works a lot better now with version 0.4.1,
`getpapers -q 'ursus maritimus' -o ursus -x` runs without problems (for 170 results).
`getpapers -q 'ursus maritimus' -o ursus -p` although gives different timeouts, as well as `getpapers -q 'ursus maritimus' -o ursus -s`. Perhaps an issue with file sizes of concurrently requested downloads?

**Comment by chreman** [comment]
I noticed most timeouts happen for papers after around nr. 160, is there a specific API-feature around that number?

**Comment by tarrow** [comment]
I am writing a patch to solve this. We have many, many open connections if we download a long url list; my patch spawns a number of workers to do this.
